<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1200</width>
    <height>800</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>900</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>伸展树可视化演示系统</string>
  </property>
  <property name="styleSheet">
   <string>
    QMainWindow {
      background-color: #f5f7fa;
      font-family: "Microsoft YaHei", "Segoe UI", sans-serif;
    }
    QPushButton {
      padding: 10px 20px;
      font-weight: bold;
      border: none;
      border-radius: 8px;
      color: white;
    }
    QPushButton:hover {
      transform: translateY(-2px);
    }
    QPushButton:pressed {
      transform: translateY(1px);
    }
    QLineEdit {
      padding: 10px 16px;
      border: 2px solid #e1e5eb;
      border-radius: 8px;
      background-color: white;
      color: #2d3436;
      font-size: 14px;
    }
    QLineEdit:focus {
      border-color: #3498db;
    }
    QTextBrowser {
      background-color: white;
      border: 2px solid #e1e5eb;
      border-radius: 8px;
      padding: 12px;
      color: #2d3436;
      font-size: 13px;
    }
   </string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <property name="spacing">
     <number>12</number>
    </property>
    <property name="leftMargin">
     <number>16</number>
    </property>
    <property name="topMargin">
     <number>16</number>
    </property>
    <property name="rightMargin">
     <number>16</number>
    </property>
    <property name="bottomMargin">
     <number>16</number>
    </property>
    <item>
     <widget class="QLabel" name="titleLabel">
      <property name="text">
       <string>伸展树可视化演示</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="styleSheet">
       <string>
        QLabel {
          font-size: 24px;
          font-weight: bold;
          color: #2c3e50;
          margin-bottom: 10px;
        }
       </string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="TreeWidget" name="treeWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>0</horstretch>
        <verstretch>1</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumHeight">
       <number>550</number>
      </property>
      <property name="styleSheet">
       <string>
        QWidget { 
          background: qlineargradient(x1:0, y1:0, x2:0, y2:1,
                                    stop:0 #ffffff, stop:1 #f8fafc);
          border: 2px solid #e3e9f3;
          border-radius: 12px;
        }
       </string>
      </property>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="controlsMainLayout">
      <property name="spacing">
       <number>10</number>
      </property>
      <item>
       <widget class="QLineEdit" name="lineEdit">
        <property name="placeholderText">
         <string>请输入数字...</string>
        </property>
        <property name="maxLength">
         <number>6</number>
        </property>
        <property name="minimumHeight">
         <number>40</number>
        </property>
        <property name="maximumWidth">
         <number>150</number>
        </property>
        <property name="font">
         <font>
          <family>Microsoft YaHei</family>
          <pointsize>11</pointsize>
         </font>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="operationButtonsLayout">
        <property name="spacing">
         <number>8</number>
        </property>
        <item>
         <widget class="QPushButton" name="insertButton">
          <property name="text">
           <string>插入</string>
          </property>
          <property name="minimumHeight">
           <number>40</number>
          </property>
          <property name="maximumWidth">
           <number>80</number>
          </property>
          <property name="font">
           <font>
            <family>Microsoft YaHei</family>
            <pointsize>11</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="styleSheet">
           <string>
            background-color: #3498db;
           </string>
          </property>
          <property name="icon">
           <iconset theme="list-add"/>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="deleteButton">
          <property name="text">
           <string>删除</string>
          </property>
          <property name="minimumHeight">
           <number>40</number>
          </property>
          <property name="maximumWidth">
           <number>80</number>
          </property>
          <property name="font">
           <font>
            <family>Microsoft YaHei</family>
            <pointsize>11</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="styleSheet">
           <string>
            background-color: #e74c3c;
           </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="searchButton">
          <property name="text">
           <string>查找</string>
          </property>
          <property name="minimumHeight">
           <number>40</number>
          </property>
          <property name="maximumWidth">
           <number>80</number>
          </property>
          <property name="font">
           <font>
            <family>Microsoft YaHei</family>
            <pointsize>11</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="styleSheet">
           <string>
            background-color: #9b59b6;
           </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="splitButton">
          <property name="text">
           <string>拆分</string>
          </property>
          <property name="minimumHeight">
           <number>40</number>
          </property>
          <property name="maximumWidth">
           <number>80</number>
          </property>
          <property name="font">
           <font>
            <family>Microsoft YaHei</family>
            <pointsize>11</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="styleSheet">
           <string>
            background-color: #f39c12;
           </string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="mergeButton">
          <property name="text">
           <string>合并</string>
          </property>
          <property name="minimumHeight">
           <number>40</number>
          </property>
          <property name="maximumWidth">
           <number>80</number>
          </property>
          <property name="font">
           <font>
            <family>Microsoft YaHei</family>
            <pointsize>11</pointsize>
            <weight>75</weight>
            <bold>true</bold>
           </font>
          </property>
          <property name="styleSheet">
           <string>
            background-color: #2ecc71;
           </string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeType">
         <enum>QSizePolicy::MinimumExpanding</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="clearButton">
        <property name="text">
         <string>清空</string>
        </property>
        <property name="minimumHeight">
         <number>40</number>
        </property>
        <property name="maximumWidth">
         <number>80</number>
        </property>
        <property name="font">
         <font>
          <family>Microsoft YaHei</family>
          <pointsize>11</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="styleSheet">
         <string>
          background-color: #e74c3c;
          border-radius: 8px;
         </string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="exitButton">
        <property name="text">
         <string>退出</string>
        </property>
        <property name="minimumHeight">
         <number>40</number>
        </property>
        <property name="maximumWidth">
         <number>80</number>
        </property>
        <property name="font">
         <font>
          <family>Microsoft YaHei</family>
          <pointsize>11</pointsize>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="styleSheet">
         <string>
          background-color: #34495e;
          border-radius: 8px;
         </string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QTextBrowser" name="textBrowser">
      <property name="maximumHeight">
       <number>100</number>
      </property>
      <property name="minimumHeight">
       <number>100</number>
      </property>
      <property name="styleSheet">
       <string>
        QTextBrowser {
          background-color: white;
          border: 2px solid #e1e5eb;
          border-radius: 8px;
          padding: 12px;
          color: #2d3436;
          font-size: 13px;
        }
       </string>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar">
   <property name="styleSheet">
    <string>
     QStatusBar {
       background-color: #f8fafc;
       color: #2c3e50;
       padding: 6px;
       font-family: "Microsoft YaHei";
       font-size: 13px;
       border-top: 1px solid #e3e9f3;
     }
    </string>
   </property>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>TreeWidget</class>
   <extends>QWidget</extends>
   <header>treewidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>